src/mudlet.cpp

```diff
@@ -953,17 +985,90 @@ void mudlet::enableToolbarButtons()
     mpMainToolBar->actions()[14]->setEnabled( true );
 }
 
-bool mudlet::openWindow( Host * pHost, const QString & name )
+bool mudlet::saveWindowLayout() {
+    qDebug() << "mudlet::saveWindowLayout() - saving layout.";
+
+    QString layoutFilePath = QStringLiteral("%1/.config/mudlet/windowLayout.dat").arg(QDir::homePath());
+
+    QFile layoutFile(layoutFilePath);
+    if (layoutFile.open(QIODevice::WriteOnly)) {
+        // revert update markers to ready objects for saving.
+        commitLayoutUpdates();
+
+        QByteArray layoutData = saveState();
+        QDataStream ofs(&layoutFile);
+        ofs << layoutData;
+        layoutFile.close();
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool mudlet::loadWindowLayout() {
+    qDebug() << "mudlet::loadWindowLayout() - loading layout.";
+
+    QString layoutFilePath = QStringLiteral("%1/.config/mudlet/windowLayout.dat").arg(QDir::homePath());
+
+    QFile layoutFile(layoutFilePath);
+    if (layoutFile.exists()) {
+        if (layoutFile.open(QIODevice::ReadOnly)) {
+            mIsLoadingLayout = true;
+
+            QByteArray layoutData;
+            QDataStream ifs(&layoutFile);
+            ifs >> layoutData;
+            layoutFile.close();
+
+            restoreState(layoutData);
```

Comment:
This can fail - and returns `true` or `false` as appropriate - I think it should be passed back up to the caller (if that is the Lua sub-system)...